master/internal/api_runs.go

```diff
@@ -507,3 +510,123 @@ func (a *apiServer) KillRuns(ctx context.Context, req *apiv1.KillRunsRequest,
 	}
 	return &apiv1.KillRunsResponse{Results: results}, nil
 }
+
+func (a *apiServer) DeleteRuns(ctx context.Context, req *apiv1.DeleteRunsRequest,
+) (*apiv1.DeleteRunsResponse, error) {
+	if len(req.RunIds) > 0 && req.Filter != nil {
+		return nil, fmt.Errorf("if filter is provided run id list must be empty")
+	}
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// get runs to delete
+	var runChecks []archiveRunOKResult
```

Comment:
similar comment to the last. maybe `deleteCandidates` is clearer? also similar comment about `archiveRunOKResult` being a strange name and the `State` field maybe being better as `is_terminal` or something.

Reply:
Updated the variable and field name. `archiveRunOKResult` is used for move, archive/unarchive & pause/unpause as well. So I renamed to something more generic: `runCandidateResult`